server/server.go

```diff
@@ -219,15 +220,13 @@ func handleHealth(w http.ResponseWriter, r *http.Request) {
 }
 
 func handleWebsocket(w http.ResponseWriter, r *http.Request) {
-	userId := getUserIdFromCookie(r)
-	responseHeader := setUserIdToCookie(w, userId)
-
-	if len(userId) == 0 {
-		userId = NewUserId()
-		responseHeader = setUserIdToCookie(w, userId)
+	userId := NewUserId()
+	if cookie, err := r.Cookie(cookiename); err == nil {
+		userId = UserId(cookie.Value)
 	}
+	http.SetCookie(w, mintCookie(userId))
```

Comment:
What is the difference between doing `http.SetCookie` and the way I was trying it (setting it on the `responseHeader`)?

Reply:
`http.SetCookie` is a sort of dual to `http.Request.Cookie` in that the former "unparses" cookies into their appropriate "Set-Cookie" header value and the latter parses "Cookie" header values into their appropriate `*http.Cookie` representation. This refactoring doesn't change any behavior; indeed we were already using `http.SetCookie(w, cookie)`. This change just omits some unnecessary dancing around with `w.Header()`